composer/trainer/trainer.py

```diff
@@ -187,9 +187,16 @@ def _get_initial_grad_accum(grad_accum: Union[int, str]):
         raise ValueError("grad_accum must be an int or ``'auto'``")
 
 
-def _get_initial_device_train_microbatch_size(device_train_microbatch_size: Union[int, str],
-                                              train_dataloader: Optional[Iterable]):
-    if device_train_microbatch_size == 'auto':
+def _get_initial_device_train_microbatch_size(device_train_microbatch_size: Optional[Union[int, str]],
+                                              auto_microbatching: bool,
+                                              train_dataloader: Optional[Iterable]) -> Optional[int]:
```

Comment:
A little confusing that the function takes `device_train_microbatch_size` and `auto_microbatching`. seems a little redundant because `device_train_microbatch_size` can be set to `'auto'`. Perhaps `_get_initial_device_train_microbatch_size` takes in `device` and then it checks like this:
```suggestion
def _get_initial_device_train_microbatch_size(device_train_microbatch_size: Optional[Union[int, str]],
                                              device: Device,
                                              train_dataloader: Optional[Iterable]) -> Optional[int]:
                                              
  if isinstance(device_train_microbatch_size, int):
    return device_train_microbatch_size
  elif _is_auto_microbatching(device_train_microbatch_size, device):
      if train_dataloader is None:
            return None
        try:
            batch_size = getattr(train_dataloader, 'batch_size')
        except AttributeError as e:
            raise AttributeError(
                'device_train_microbatch_size requires the `state.train_dataloader` to have a `batch_size` attribute.'
            ) from e
        return batch_size
        
  else:
    raise ValueError("device_train_microbatch_size must be an int or ``'auto'``")
                                        
```